### PR TITLE
memcached: Handle unexpected network timeout errors

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

In addtition to `memcache: connect timeout to XXX.XXX.XXX.XXX:11211`, we have been seeing `read tcp XXX.XXX.XXX.XXX:59812->XXX.XXX.XXX.XXX:11211: i/o timeout`. This PR aims to address them.

## Changes

* Handle network timeout errors.

## Verification

* `make test-local`

### CODE REVIEW

File: pkg/cacheutil/memcached_client.go [hunk 1]
```diff
@@ -492,6 +473,27 @@ func (c *memcachedClient) getMultiSingle(ctx context.Context, keys []string) (it
...
+	var connErr *memcache.ConnectTimeoutError
+	var netErr net.Error
+	switch {
+	case errors.As(err, &connErr):
+		c.failures.WithLabelValues(op, reasonTimeout).Inc()
+	case errors.As(err, &netErr):
+		if netErr.Timeout() {
+			c.failures.WithLabelValues(op, reasonTimeout).Inc()
+		} else {
+			c.failures.WithLabelValues(op, reasonOther).Inc()
```
Comment (kakkoyun) [inline pkg/cacheutil/memcached_client.go:486]: I can introduce another label to cover `net.Error`. WDYT?

Comment (metalmatze) [reply pkg/cacheutil/memcached_client.go:486]: Sure. Doesn't sound too bad in terms of cardinality. Is there a specific reason we would need it right now?

Comment (kakkoyun) [reply pkg/cacheutil/memcached_client.go:486]: Noop. That's why I went with other. I just need to catch the timeout errors.

Comment (pracucci) [reply pkg/cacheutil/memcached_client.go:486]: Agree having a dedicated label value would be great here.

Comment (brancz) [reply pkg/cacheutil/memcached_client.go:486]: I don't feel this is problematic, it seems only helpful

File: pkg/cacheutil/memcached_client.go [hunk 2]
```diff
@@ -492,6 +473,27 @@ func (c *memcachedClient) getMultiSingle(ctx context.Context, keys []string) (it
 	return items, err
 }
 
+func (c *memcachedClient) handleError(op string, err error) {
```
Comment (pracucci) [inline pkg/cacheutil/memcached_client.go:476]: Maybe "trackError" is more accurate then "handleError", given it just track metrics?

**Pull request merged**